src/bitset.ml

```diff
@@ -0,0 +1,224 @@
+module Make (B : Context.A_DISK) = struct
+  module Sector = Sector.Make (B)
+  module Schema = Schema.Make (B)
+
+  type t = Sector.t
+
+  open Lwt_result.Syntax
+
+  let get_page_size () = B.page_size
+
+  let get_nb_leaves () =
+    let nb_sectors = Int64.to_int B.nb_sectors in
+    let page_size = get_page_size () in
+    let bit_size = page_size * 8 in
+    (nb_sectors + bit_size - 1) / bit_size
+
+  let get_group_size nb_children nb_leaves =
+    let rec get_group_size group_size =
+      if group_size * nb_children >= nb_leaves
+      then group_size
+      else get_group_size (group_size * nb_children)
+    in
+    get_group_size 1
+
+  let get_ptr_size () =
+    let pointer_size = Sector.ptr_size in
+    let id_size = Sector.id_size in
+    ((pointer_size + id_size) / 8) + 8
```

Comment:
This looks complicated! (and I'm afraid it'll break with different disk sizes) Any reasons why `Sector.ptr_size` isn't enough? :)